### ISSUE TITLE
$XDG_RUNTIME_DIR must have 0700 as permissions

### ISSUE THREAD

**Issue by GeorgeBekh** [WerWolv/ImHex#836]
### Operating System

Linux

### What's the issue you encountered?

Run:
`flatpak run net.werwolv.ImHex`

Output:
```
[18:31:14] [INFO]  [main] Welcome to ImHex 1.25.0!
[18:31:14] [INFO]  [main] Native scaling set to: 1.2
[18:31:14] [ERROR] [main] Init task 'Creating directories' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[18:31:14] [ERROR] [main] Failed to load configuration! $XDG_RUNTIME_DIR must have 0700 as permissions
[18:31:14] [ERROR] [main] Init task 'Loading settings' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[18:31:14] [ERROR] [main] Init task 'Loading plugins' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[18:31:14] [INFO]  [main] Task 'Checking for updates' finished in 0 ms
[18:31:17] [INFO]  [main] Task 'Loading fonts' finished in 2750 ms
```

### How can the issue be reproduced?

install ImHex from flatpak and run it

### ImHex Version

1.25.0

### ImHex Build Type

- [ ] Nightly or built from sources

### Installation type

flatpak

**Comment by LordGaav** [comment]
I can also reproduce this with the Flatpak version of ImHex 1.26.2 on Debian 11:

```
lordgaav@asura:~$ flatpak run net.werwolv.ImHex
[12:59:52] [INFO]  [main] Welcome to ImHex 1.26.2!
[12:59:52] [INFO]  [main] Native scaling set to: 1.0
[12:59:52] [INFO]  [main] Task 'Setting up environment' finished in 0 ms
[12:59:52] [ERROR] [main] Init task 'Creating directories' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[12:59:52] [ERROR] [main] Failed to load configuration! $XDG_RUNTIME_DIR must have 0700 as permissions
[12:59:52] [ERROR] [main] Init task 'Loading settings' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[12:59:52] [ERROR] [main] Init task 'Loading plugins' threw an exception: $XDG_RUNTIME_DIR must have 0700 as permissions
[12:59:52] [INFO]  [main] Task 'Checking for updates' finished in 0 ms
[12:59:54] [INFO]  [main] Task 'Loading fonts' finished in 1794 ms
terminate called after throwing an instance of 'xdg::BaseDirectoryException'
  what():  $XDG_RUNTIME_DIR must have 0700 as permissions

lordgaav@asura:~$ echo $XDG_RUNTIME_DIR
/run/user/1000

lordgaav@asura:~$ ls -la /run/user
total 0
drwxr-xr-x  3 root     root      60 14 jan 19:05 .
drwxr-xr-x 38 root     root     980 15 jan 12:03 ..
drwx------ 18 lordgaav lordgaav 540 15 jan 12:55 1000
```

The dir already has permissions 0700, but ImHex is convinced it's not.

**Comment by LordGaav** [comment]
Problem was solved for me by updating to Flatpak 1.14.1-1~bpo11+1 from the bullseye backports. Seems like an issue fixed in Flatpak 1.11.1: https://github.com/flatpak/flatpak/issues/3397

**Comment by Stadtmusiker** [comment]
Similar errors for NixOS (stable `23.11`). I will reach out Nix team, backlink them here.

```
[nix-shell:~]$ nix-shell -p imhex
# (stdout stuff...)

[nix-shell:~]$ nix-info
system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.18.1, channels(root): "nixos-23.11, nixos-wsl", nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos

[nix-shell:~]$ echo $XDG_RUNTIME_DIR
/run/user/1000/

[nix-shell:~]$ ls -l $XDG_RUNTIME_DIR/..
total 0
drwxr-xr-x 5 nixos users 180 Mar 18 21:06 1000

[nix-shell:~]$ imhex
terminate called after throwing an instance of 'xdg::BaseDirectoryException'
  what():  $XDG_RUNTIME_DIR must have 0700 as permissions
Aborted
```

Also tried `nixos-unstable` channel today, same behavior.
